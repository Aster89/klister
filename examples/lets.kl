#lang kernel

(example (the Syntax (let (x 'foo) x)))

(example (flet (f (x) (if x 'a 'b)) (f #true)))

(example (the Syntax (flet (f (x) 'a) (list-syntax ((f #true) (f 4)) 'here))))
