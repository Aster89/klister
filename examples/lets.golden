#[lets.kl:3.33-3.36]<foo> : Syntax
#[lets.kl:5.40-5.41]<a> : Syntax
#[lets.kl:7.74-7.78]<(a a)> : Syntax
#[lets.kl:9.82-9.86]<(a a)> : Syntax
#[lets.kl:13.29-13.32]<foo> : Syntax
45 : Integer
